### PR TITLE
6070: sealed contact addresses for unprivileged users

### PR DESCRIPTION
**Pull request by kkoskelin**

### CODE REVIEW

File: shared/src/business/useCases/getCaseInteractor.test.js [hunk 1]
```diff
@@ -150,6 +155,51 @@ describe('Get case', () => {
     ).rejects.toThrow('Unauthorized');
   });
 
+  describe('access to contact information which is sealed', () => {
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/getCaseInteractor.test.js:158]: 👏

Comment (rachelschneiderman) [reply shared/src/business/useCases/getCaseInteractor.test.js:158]: Everything about these tests are awesome, the setup, the test naming.... 👏

**Review by rachelschneiderman — APPROVED** [review]

File: shared/src/business/utilities/caseFilter.test.js [hunk 2]
```diff
@@ -27,19 +27,26 @@ describe('caseFilter', () => {
...
     const result = caseContactAddressSealedFormatter({
-      bananas: '12',
+      additionalName: 'Bob',
+      bananas: '8-foot bunch',
       city: 'Los Angeles',
       contactId: '42-universe-everything',
+      inCareOf: 'Friendship is Magic',
       isAddressSealed: 'maybe',
       name: 'Joe Dirt',
+      otherFilerType: 'Nail File',
```
Comment (mrinsin) [inline shared/src/business/utilities/caseFilter.test.js:37]: LOL

File: shared/src/business/useCases/getCaseInteractor.js [hunk 3]
```diff
@@ -119,5 +122,38 @@ exports.getCaseInteractor = async ({ applicationContext, docketNumber }) => {
...
+    if (
+      caseDetailRaw.contactSecondary &&
+      caseDetailRaw.contactSecondary.isAddressSealed
+    ) {
+      caseDetailRaw.contactSecondary = caseContactAddressSealedFormatter(
+        caseDetailRaw.contactSecondary,
+      );
+    }
+
+    if (Array.isArray(caseDetailRaw.otherFilers)) {
```
Comment (mrinsin) [inline shared/src/business/useCases/getCaseInteractor.js:149]: I think we will need to do this for `otherPetitioners` too (which is different from `otherFilers`, I just learned!

File: shared/src/business/useCases/getCaseInteractor.test.js (hunk 1 above)
Comment (mrinsin) [reply shared/src/business/useCases/getCaseInteractor.test.js:158]: +1

**Review by mrinsin — APPROVED** [review]
Awesome stuff!

**Pull request merged**